web/client/utils/ConfigUtils.js

```diff
@@ -50,6 +50,38 @@ const getConfigurationOptions = function(query, defaultName, extension, geoStore
     };
 };
 
+/**
+ * it removes some params from the query string
+ * return the shrinked url
+*/
+const removeParameters = (urlToFilter, skip) => {
+    const urlparts = urlToFilter.split('?');
+    let paramsFiltered = "";
+    if (urlparts.length >= 2 && urlparts[1]) {
+        const pars = urlparts[1].split(/[&;]/g).filter( p => !!p);
+        pars.forEach((par, i) => {
+            const param = par.split('=');
+            if (skip.indexOf(param[0].toLowerCase()) === -1) {
+                let addAnd = i === (pars.length - 1) ? "" : "&";
+                paramsFiltered += param.join("=") + addAnd;
+            }
+        });
+    }
+    return !!paramsFiltered ? urlparts[0] + "?" + paramsFiltered : urlparts[0];
+};
+/**
+ * WORKAROUND: it removes the extra ? when the authkey param is present
+ * the url was like         http......?authkey=....?service=....&otherparam
+ * that should become like  http......?authkey=....&service=....&otherparam
+*/
+const normalizeUrl = (urlToNormalize) => {
```

Comment:
the name doesn't remind what you are doing. Call it something like `cleanDuplicatedQuestionMarks` and normalizeUrl can be the combination of removeParameters + cleanDuplicatedQuestionMarks (I've seen you used it often)